File: remote/src/main/java/com/kelsos/mbrc/dao/Album.java
package com.kelsos.mbrc.dao;

import com.kelsos.mbrc.RemoteDatabase;
import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.ForeignKey;
import com.raizlabs.android.dbflow.annotation.ForeignKeyReference;
import com.raizlabs.android.dbflow.annotation.PrimaryKey;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.structure.BaseModel;

@Table(databaseName = RemoteDatabase.NAME) public class Album extends BaseModel {
  @Column @PrimaryKey(autoincrement = true) long id;
  @Column private String name;
  @Column @ForeignKey(
      references = {
          @ForeignKeyReference(columnName = "artist_id",
              columnType = Long.class,
              foreignColumnName = "id")
      }, saveForeignKeyModel = false) private Artist artist;

  @Column @ForeignKey(
      references = {
          @ForeignKeyReference(columnName = "cover_id",
              columnType = Long.class,
              foreignColumnName = "id")
      }, saveForeignKeyModel = false) private Cover cover;

  public long getId() {
    return id;
  }

  public Album setId(long id) {
    this.id = id;
    return this;
  }

  public String getName() {
    return name;
  }

  public Album setName(String name) {
    this.name = name;
    return this;
  }

  public Artist getArtist() {
    return artist;
  }

  public Album setArtist(Artist artist) {
    this.artist = artist;
    return this;
  }

  public Cover getCover() {
    return cover;
  }

  public Album setCover(Cover cover) {
    this.cover = cover;
    return this;
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/data/dao/Album.kt
package com.kelsos.mbrc.data.dao

import com.kelsos.mbrc.data.db.RemoteDatabase
import com.raizlabs.android.dbflow.annotation.*
import com.raizlabs.android.dbflow.kotlinextensions.modelAdapter
import com.raizlabs.android.dbflow.structure.Model

@Table(database = RemoteDatabase::class,
    name = "albums",
    indexGroups = arrayOf(IndexGroup(number = 1, name = "album_name_index")))
data class Album(
    @Column(name = "name") @Index(indexGroups = intArrayOf(1)) var name: String = "",
    @Column
    @ForeignKey(references = arrayOf(ForeignKeyReference(columnName = "artist_id",
        foreignKeyColumnName = "id")), saveForeignKeyModel = false)
    var artist: Artist? = null,
    @Column
    @ForeignKey(references = arrayOf(ForeignKeyReference(columnName = "cover_id",
        foreignKeyColumnName = "id")), saveForeignKeyModel = false)
    var cover: Cover? = null,
    @Column(name = "date_added") var dateAdded: Long = 0,
    @Column(name = "date_updated") var dateUpdated: Long = 0,
    @Column(name = "date_deleted") var dateDeleted: Long = 0,
    @PrimaryKey(autoincrement = true) var id: Long = 0) : Model {

  override fun load() = modelAdapter<Album>().load(this)

  override fun insert(): Long = modelAdapter<Album>().insert(this)

  override fun save() = modelAdapter<Album>().save(this)

  override fun update() = modelAdapter<Album>().update(this)

  override fun exists(): Boolean = modelAdapter<Album>().exists(this)

  override fun delete() = modelAdapter<Album>().delete(this)
}
